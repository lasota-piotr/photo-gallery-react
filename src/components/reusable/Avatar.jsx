import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const AvatarImg = styled.img`
  width: ${props => props.theme.unitXxlarge};
  height: ${props => props.theme.unitXxlarge};
  border-radius: 50%;
  overflow: hidden;
  object-fit: cover;
`;

const Avatar = ({ name, src }) => <AvatarImg src={src} alt={name} />;

Avatar.propTypes = {
  name: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
};

export default Avatar;
