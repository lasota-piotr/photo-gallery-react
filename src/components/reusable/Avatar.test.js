import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from './Avatar';

test('Generates Avatar', () => {
  const component = renderer.create(
    <Avatar name="John" src="example.com.jpg" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
