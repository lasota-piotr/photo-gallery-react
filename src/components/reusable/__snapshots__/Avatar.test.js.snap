// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates Avatar 1`] = `
<img
  alt="John"
  className="sc-bdVaJa kPDVQV"
  src="example.com.jpg"
/>
`;
